integration-cli/docker_cli_push_test.go

```diff
@@ -111,9 +115,7 @@ func (s *DockerSuite) TestPushInterrupt(c *check.C) {
 	if err := pushCmd.Process.Kill(); err != nil {
 		c.Fatalf("Failed to kill push process: %v", err)
 	}
-	// Try agin
-	pushCmd = exec.Command(dockerBinary, "push", repoName)
-	if out, err := pushCmd.CombinedOutput(); err == nil {
+	if out, err := d.Cmd("push", repoName); err == nil {
```

Comment:
I would prefer if we didn't exit this test until the push finished. So rather than starting up a new daemon, let's just wait until this push is done (ie. we don't get back "already in progress" error, after the 1st one) and then let the func exit.